scipy/interpolate/_rgi.py

```diff
@@ -243,6 +249,14 @@ def __init__(self, points, values, method="linear", bounds_error=True,
         self.fill_value = self._check_fill_value(self.values, fill_value)
         if self._descending_dimensions:
             self.values = np.flip(values, axis=self._descending_dimensions)
+        
+        if self.method == "pchip" and np.iscomplexobj(self.values):
+            msg = ("`PchipInterpolator` only works with real values. Passing "
+                   "complex-dtyped `values` with `method='pchip'` is deprecated "
+                   "and will raise an error in SciPy 1.15.0. If you are trying to "
+                   "use the real components of the passed array, use `np.real` on "
+                   "the array before passing to `RegularGridInterpolator`.")
+            warnings.warn(msg, DeprecationWarning, stacklevel=2)
```

Comment:
What do we want the `stacklevel` to be?

Reply:
Do you know if we need a deprecation warning here? `RegularGridInterpolator` calls eventually calls `PchipInterpolator` so with this, I think you'd end up with two sets of warnings (although I haven't tried it!)

Reply:
Ah yes, there would be two sets of warnings. I thought it would be useful to have a separate message here to be clear that the issue is with `method='pchip'`, but probably not needed.